### PR TITLE
Pool create command.

### PR DESCRIPTION
**Pull request by anastasiamac**
Create pool command, plugged in + tests.

### CODE REVIEW

File: api/storage/client_test.go [hunk 1]
```diff
@@ -102,6 +102,8 @@ func (s *storageMockSuite) TestListPools(c *gc.C) {
 			}
 
 			return nil
+			//	"github.com/juju/juju/apiserver/params"
```
Comment (wallyworld) [inline api/storage/client_test.go]: delete me

File: apiserver/storage/pool_create_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	jujustorage "github.com/juju/juju/storage"
+	"github.com/juju/juju/storage/pool"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/apiserver/storage"
+	"github.com/juju/juju/apiserver/testing"
+	jujutesting "github.com/juju/juju/juju/testing"
+)
+
+type poolCreateSuite struct {
```
Comment (wallyworld) [inline apiserver/storage/pool_create_test.go]: poolcreate_test.go would be a more conventional name

File: apiserver/storage/storage.go [hunk 3]
```diff
@@ -167,3 +168,33 @@ func filterPoolInstance(typeSet, nameSet set.Strings, apool pool.Pool) (params.S
 	}
 	return one, true
 }
+
+func (a *API) CreatePool(p params.StoragePool) error {
+	providerT := storage.ProviderType(p.Type)
```
Comment (wallyworld) [inline apiserver/storage/storage.go]: add a todo to move this to a business layer

File: api/storage/client_test.go (hunk 1 above)
Comment (anastasiamac) [inline api/storage/client_test.go]: how did this get here? of course, I'll delete :D

File: cmd/juju/storage/pool_create.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 Canonical Ltd.
```
Comment (wallyworld) [inline cmd/juju/storage/pool_create.go]: poolcreate.go please (don't necessarily agree, just convention)

File: cmd/juju/storage/pool_create.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	PoolCommandBase
+	pname string
+	ptype string
+	pcfg  map[string]interface{}
+}
+
+// Init implements Command.Init.
+func (c *PoolCreateCommand) Init(args []string) (err error) {
+	if c.ptype == "" {
+		return fmt.Errorf("no provider type for pool specified")
```
Comment (wallyworld) [inline cmd/juju/storage/pool_create.go]: add todo that this will need to become optional

**Comment by wallyworld** [comment]
Just a few minor quibbles, tyvm

File: apiserver/storage/pool_create_test.go (hunk 2 above)
Comment (anastasiamac) [inline apiserver/storage/pool_create_test.go]: done

File: apiserver/storage/storage.go (hunk 3 above)
Comment (anastasiamac) [inline apiserver/storage/storage.go]: added todo and renamed variable

File: cmd/juju/storage/pool_create.go (hunk 4 above)
Comment (anastasiamac) [inline cmd/juju/storage/pool_create.go]: done

File: cmd/juju/storage/pool_create.go (hunk 5 above)
Comment (anastasiamac) [inline cmd/juju/storage/pool_create.go]: done

**Pull request merged**